src/backend/optimizer/path/allpaths.c

```diff
@@ -2125,15 +2125,38 @@ set_cte_pathlist(PlannerInfo *root, RelOptInfo *rel, RangeTblEntry *rte)
 			subplan = subquery_planner(cteroot->glob, subquery, cteroot, cte->cterecursive,
 									   tuple_fraction, &subroot, config);
 
-			cteplaninfo->shared_plan = prepare_plan_for_sharing(cteroot, subplan);
+			if (subplan->flow->locustype != CdbLocusType_SegmentGeneral)
```

Comment:
maybe it's better this way?
```suggestion
			if (CdbPathLocus_IsSegmentGeneral(*subplan->flow))
```